aten/src/ATen/gen_backend_select_register.py

```diff
@@ -24,12 +24,27 @@
 GENERATED_COMMENT = CodeTemplate(
     "@" + "generated from ${filename}")
 
-FUNCTION_REGISTRATION = CodeTemplate("""\
+UNBOXEDONLY_FUNCTION_REGISTRATION = CodeTemplate("""\
   m.impl_UNBOXED("aten::${op_name_with_overload_name}", ${function_name});
 """)
 
+FUNCTION_REGISTRATION = CodeTemplate("""\
```

Comment:
Same suggestion re making hacky_wrapper conditional. Could use the same central discriminant function to decide; this would actually reinforce the difference as a canonical part of codegen logic and clarify where/why it was necessary.